specs/error_state/ErrorOutOfGasAccountAccess.md

```diff
@@ -0,0 +1,41 @@
+# ErrorOutOfGasAccountAccess state
+
+## Procedure
+
+Handle the corresponding out of gas errors for `BALANCE`, `EXTCODESIZE`, and `EXTCODEHASH` opcodes which possibly touches an extra account.
```

Comment:
Not sure what "possibly touches" mean? Does it refer only to `EXTCODESIZE`? Are there some special constraints to be added when it touches an extra account?

Reply:
My understanding is all of these three opcodes could touch an external account and we handle an account access [here](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/476/files#diff-67ee79e787a7b6bbf6a48ab2f1dbf05f79b5d5fe69fe615b238277f59eff9f6dR32-R33), 
```rust
    is_warm = instruction.read_account_to_access_list(tx_id, address)
    gas_cost = GAS_COST_WARM_ACCESS if is_warm == FQ(1) else GAS_COST_ACCOUNT_COLD_ACCESS
```

If these opcode access a non-external account, it must be a warm access. So the above snippet is to take care `possibly touches an extra account`.

Reply:
Thanks Kimi. I didn't know this refers to the warm access. I think it's somehow misleading, because it's not very clear what "extra" means. I think the part "which possibly touches an extra account" could be simply removed.

Also, I don't think it needs to be warm access in some specific cases - it can be warm or cold, just the price is different then, no?

Reply:
I was thinking to revise to "which possibly touches an `external` account". Will it be better?

> Also, I don't think it needs to be warm access in some specific cases

I didn't get it. Could you elaborate more?

> it can be warm or cold, just the price is different then, no?

Yes, just the price differences.

Reply:
> I was thinking to revise to "which possibly touches an `external` account". Will it be better?

Hm, I just don't understand what is `external` account?

> 
> > Also, I don't think it needs to be warm access in some specific cases
> 
> I didn't get it. Could you elaborate more?

I was referring to "If these opcode access a non-external account, it must be a warm access."

> 
> > it can be warm or cold, just the price is different then, no?
> 
> Yes, just the price differences.